test/integration/api/datasets.js

```diff
@@ -3954,6 +3982,68 @@ describe('datasets and entities', () => {
         }));
       });
 
+      describe('dataset properties name conflicts via Form upload', () => {
+        it('should reject if property name differ by just capitalization', testService(async (service) => {
+          const alice = await service.login('alice');
+
+          // dataset "people" with property "first_name"
+          await alice.post('/v1/projects/1/forms?publish=True')
+            .send(testData.forms.simpleEntity)
+            .set('Content-Type', 'application/xml')
+            .expect(200);
+
+          // dataset "people" with property "FIRST_NAME"
+          await alice.post('/v1/projects/1/forms?publish=True')
```

Comment:
What happens if ?publish=True is not specified (if you're uploading a draft, but not publishing)? Am I right in thinking that that's also a 409, since `first_name` is a published property at this point? Either way, would you mind adding a request along those lines to this test just to document the behavior?

Reply:
I see that 75d0de3f810e375fca8bf62415581a6ed985f3b3 adds a test along these lines. Looks good!